ulib/experimental/FStar.OrdSet.fsti

```diff
@@ -145,59 +181,158 @@ val fold (#a:eqtype) (#acc:Type) (#f:cmp a) (g:acc -> a -> acc) (init:acc) (s:or
 val map (#a #b:eqtype) (#fa:cmp a) (#fb:cmp b) (g:a -> b) (sa:ordset a fa)
   : Pure (ordset b fb)
     (requires (forall x y. x `fa` y ==> g x `fb` g y))
-    (ensures (fun sb -> True))
+    (ensures (fun sb -> (size sb <= size sa) /\                   
```

Comment:
Can we can improve this map spec, e.g. if we also have a precondition that `g` is injective, we can prove that `as_list sb == List.Tot.map g (as_list sa)`. The current spec only talks about head of the returned set, and nothing about the rest of the set.

Reply:
We most certainly can and should. Will emit additional update soon.